src/systemctl/systemctl.c

```diff
@@ -5403,6 +5404,22 @@ static int print_property(const char *name, const char *expected_value, sd_bus_m
                         if (all || !isempty(fields))
                                 bus_print_property_value(name, expected_value, value, strempty(fields));
 
+                        return 1;
+                } else if (contents[0] == SD_BUS_TYPE_BYTE && streq(name, "CPUAffinity")) {
+                        _cleanup_free_ char *mask = NULL;
+                        const void *a;
+                        size_t n;
+
+                        r = sd_bus_message_read_array(m, 'y', &a, &n);
+                        if (r < 0)
+                                return bus_log_parse_error(r);
+
+                        cpu_set_from_dbus((cpu_set_t *)a, n);
+
+                        mask = cpu_set_to_string(a, n);
+                        if (mask)
+                                bus_print_property_value(name, expected_value, value, mask);
+
```

Comment:
IIRC the original idea behind systemctl show verb was to present properties of the unit in format that is easily usable in shell scripts.

Hence, previous output was borked and not helpful since it basically was just a huuuge number. Practically speaking my proposed output can't be mistaken with the old output.

As for ranges, I think that simple list of CPU indices is a better output considering the use case, which is arguably shell scripting. For example, from the simple list you can directly create bash array and work with easily.

Also on RHEL/CentOS side I've only encountered use cases where people want to isolate services from each other. Hence CPUAffinity mask usually contains only single CPU. So implementing ranges would make output just ugly for the common case (I mean both {1} and 1-1 would be pretty confusing).

I think ranges could be implemented as part of systemctl status output.